packages/elastic-apm-synthtrace/src/scripts/examples/04_anomaly.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { range } from 'lodash';
+import { apm, timerange } from '../../index';
+import { Scenario } from '../scenario';
+import { RunOptions } from '../utils/parse_run_cli_flags';
+
+/**
+ * How to run it:
+ * node packages/elastic-apm-synthtrace/src/scripts/run
+ * packages/elastic-apm-synthtrace/src/scripts/examples/04-anomaly.ts
+ * --target=ES_HOST
+ * --username USERNAME
+ * --password PWD
+ * --from now-2w
+ * --live
+ *
+ * It's important to have enough data before to generate an anomaly, so set --from now-2w
+ * @param runOptions
+ * @returns
+ */
+const scenario: Scenario = async (runOptions: RunOptions) => {
+  return {
+    generate: ({ from, to }, options) => {
+      const serviceC = apm.service('My Service', 'production', 'java').instance('a');
+      const NORMAL_DURATION = 100;
+      const NORMAL_RATE = 1;
+      let firstRun = options?.isLiveMode;
+
+      const events = timerange(new Date(from).getTime(), new Date(to).getTime())
+        .interval('1m')
+        .rate(1)
+        .spans((timestamp) => {
+          // randomly spikes to keep creating anomalies
+          const isInSpike = options?.isLiveMode && Math.random() * 100 < 2;
```

Comment:
Why do we need to support live anomalies?

Reply:
the anomalies alert only fires up if there's an anomaly within the last 30 minutes, that's why I thought would be good to support live anomalies

Reply:
I think a better approach for this is an E2E test - do you mind adding one?

Reply:
I can add an E2E test, but I think it's a good idea to keep this example for future changes on the anomaly rule. It makes it very easy to test it since we cannot rely on our cluster for this kind of test.

Reply:
Let's add an E2E test regardless. I don't think this is a good idea, because we're essentially asking whoever uses this to wait until this scenario randomly generates an anomaly. I think it'll end up confusing folks. I'd prefer for this scenario to not think about live mode or not and to always generate a spike at a fixed interval. Since ML jobs run with a lookback of 4 weeks that should yield consistent results.

Reply:
just so you know, the first spike will always happen, which creates an anomaly and changes the health status to `critical`, the live mode will keep spiking randomly to be able to test the rules.